core/vmpu/inc/vmpu.h

```diff
@@ -161,7 +161,8 @@ extern int vmpu_fault_recovery_bus(uint32_t pc, uint32_t sp, uint32_t fault_addr
 
 uint32_t vmpu_fault_find_acl(uint32_t fault_addr, uint32_t size);
 
-extern void vmpu_acl_stack(uint8_t box_id, uint32_t bss_size, uint32_t stack_size);
+extern void vmpu_acl_stack(uint8_t box_id, uint32_t bss_size, uint32_t stack_size, uint32_t * bss_start,
+                           uint32_t * stack_pointer);
```

Comment:
I don't think this is necessary as the state is now set exactly where the code has been moved to:

```C
    uint32_t bss_start = 0;
    uint32_t stack_pointer = 0;
    if (box_id == 0) {
        /* The public box does not need an explicit ACL. Instead, it uses the
         * default access permissions given by the background regions. */
        /* Note: This relies on the assumption that all supported MPU
         *       architectures allow us to setup a background region somehow. */
        bss_start = (uint32_t) __uvisor_config.heap_start;       <----------------- Previously using the state
        stack_pointer = __get_PSP();
    } else {
        uint32_t stack_size = box_cfgtbl->stack_size;
        vmpu_acl_stack(box_id, bss_size, stack_size, &bss_start, &stack_pointer);
    }

    /* Set the box state for the SRAM sections. */
    g_context_current_states[box_id].bss = bss_start;            <----------------- State now set here
    g_context_current_states[box_id].bss_size = bss_size;
    g_context_current_states[box_id].sp = stack_pointer;
```

so *technically* it is using the state (because the same local variable will be used to setup the state few lines later) (see `stack_pointer` for example).